File: ios/streaming/cache/HlsCacheBudget.swift
import Foundation

enum HlsCacheBudget {
  static let defaultMaxBytes = 4 * 1_024 * 1_024 * 1_024
  static let minimumMaxBytes = 64 * 1_024 * 1_024

  static func normalize(_ bytes: Int) -> Int {
    max(minimumMaxBytes, bytes)
  }

  static func evictionTarget(for maxBytes: Int) -> Int {
    max(minimumMaxBytes, maxBytes * 90 / 100)
  }
}

File: android/src/main/java/com/nitroplay/streaming/cache/HlsCacheBudget.kt
package com.nitroplay.hls

object HlsCacheBudget {
    const val DEFAULT_MAX_BYTES: Long = 4L * 1024L * 1024L * 1024L
    const val MINIMUM_MAX_BYTES: Long = 64L * 1024L * 1024L

    fun normalize(bytes: Long): Long = bytes.coerceAtLeast(MINIMUM_MAX_BYTES)

    fun evictionTarget(maxBytes: Long): Long = (maxBytes * 90 / 100).coerceAtLeast(MINIMUM_MAX_BYTES)
}
